Add TypeScript

grebaldi opened the pull request:
This PR adds typescript to the frontend dev setup of Sitegeist.Monocle and translates all JavaScript sources to typescript 1:1 (as far as possible).

The typings are very rough and all over the place, but it's fine for now. Any further optimizations should be addressed in future refactorings.

mficzel commented:
Looking nice but i stumbled a bit and could not complete the review:

> "invalid_type at ui.localePresets Expected object, received null" 
I think localePresets should be nullable

>  "invalid_type at styleguideObjects Expected object, received array" 
Happens when there are no items in the styleguide, a message ("add @styleguide annotations to your prototypes would be better".

Once those are fixed i can continue testing

grebaldi commented:
Both issues are fixed now. This was due to a more strict validation of the API response. It is possible that there are more cases like this, but with these 2 now fixed and the ones I already ran into and fixed right away (I tested with a full configuration), I think it should be fine now. But more testing never hurts :)

mficzel commented:
Looks nice and works. I could not find any more nullability issues. Totally agree that those are to be expected when type-safety is added.

Would love to do a screenshare to understand this a ts-thingy bit deeper.
